esphome/components/modbus_controller/output/modbus_output.cpp

```diff
@@ -51,11 +49,62 @@ void ModbusOutput::write_state(float value) {
   parent_->queue_command(write_cmd);
 }
 
-void ModbusOutput::dump_config() {
+void ModbusFloatOutput::dump_config() {
   ESP_LOGCONFIG(TAG, "Modbus Float Output:");
   LOG_FLOAT_OUTPUT(this);
-  ESP_LOGCONFIG(TAG, "Modbus device start address=0x%X register count=%d value type=%hhu", this->start_address,
-                this->register_count, this->sensor_value_type);
+  ESP_LOGCONFIG(TAG, "  Device start address=0x%X", this->start_address);
+  ESP_LOGCONFIG(TAG, "  Register count=0x%d", this->register_count);
+  ESP_LOGCONFIG(TAG, "  Value type=0x%d", static_cast<int>(this->sensor_value_type));
```

Comment:
Using `0x%d` makes no sense, as you output decimal while presenting it as hex. Furthermore, this is not the style we use for config dumps (I know it's nitpicky, but having things like these consistent really makes it all feel much more polished and less half-done). 
```suggestion
  ESP_LOGCONFIG(TAG, "  Device start address: 0x%X", this->start_address);
  ESP_LOGCONFIG(TAG, "  Register count: %d", this->register_count);
  ESP_LOGCONFIG(TAG, "  Value type: %d", static_cast<int>(this->sensor_value_type));
```

Reply:
fixed